src/undofeature.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import Feature from '../feature.js';
+import UndoCommand from './undocommand.js';
+
+/**
+ * Undo feature.
+ *
+ * Undo features brings in possibility to undo and re-do changes done in Tree Model by deltas through Batch API.
+ */
+export default class UndoFeature extends Feature {
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * Undo command which manages undo {@link engine.treeModel.Batch batches} stack (history).
+		 * Created and registered during {@link undo.UndoFeature#init feature initialization}.
+		 *
+		 * @private
+		 * @member {undo.UndoCommand} undo.UndoFeature#_undoCommand
+		 */
+		this._undoCommand = null;
+
+		/**
+		 * Undo command which manages redo {@link engine.treeModel.Batch batches} stack (history).
+		 * Created and registered during {@link undo.UndoFeature#init feature initialization}.
+		 *
+		 * @private
+		 * @member {undo.UndoCommand} undo.UndoFeature#_redoCommand
+		 */
+		this._redoCommand = null;
+
+		/**
+		 * Keeps track of which batch has already been added to undo manager.
+		 *
+		 * @private
+		 * @member {WeakSet.<engine.treeModel.Batch>} undo.UndoFeature#_batchRegistry
+		 */
+		this._batchRegistry = new WeakSet();
+	}
+
+	/**
+	 * Initializes undo feature.
+	 */
+	init() {
+		// Create commands.
+		this._redoCommand = new UndoCommand( this.editor );
```

Comment:
Aren't there any problems with having access to editor configuration or plugins loaded on demand? I have a feeling that we already discussed it and agreed that initialization other than settings values of class properties should be done in `init`.